test/e2e/far_e2e_test.go

```diff
@@ -244,26 +266,71 @@ func buildNodeParameters(clusterPlatformType configv1.PlatformType) (map[v1alpha
 	return testNodeParam, nil
 }
 
-// randomizeWorkerNode returns a worker node that his name is different than the previous one
-// (on the first call it will allways return new node)
-func randomizeWorkerNode(nodes *corev1.NodeList) *corev1.Node {
+// getAvailableNodes a list of available worker nodes in the cluster
+func getAvailableWorkerNodes() *corev1.NodeList {
+	availableNodes := &corev1.NodeList{}
+	selector := labels.NewSelector()
+	requirement, _ := labels.NewRequirement(medik8sLabels.WorkerRole, selection.Exists, []string{})
+	selector = selector.Add(*requirement)
+	Expect(k8sClient.List(context.Background(), availableNodes, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
+	if len(availableNodes.Items) < 1 {
+		Fail("No worker nodes found in the cluster")
+	}
+	return availableNodes
+}
+
+// pickRemediatedNode randomly returns a next remediated node from the current available nodes,
+// and then the node is removed from the list of available nodes
+func pickRemediatedNode(availableNodes *corev1.NodeList) *corev1.Node {
+	if len(availableNodes.Items) < 1 {
+		Fail("No available node found for remediation")
+	}
 	// Generate a random seed based on the current time
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	// Randomly select a worker node
-	return &nodes.Items[r.Intn(len(nodes.Items))]
+	selectedNodeIndex := r.Intn(len(availableNodes.Items))
+	// Delete the selected node from the list of available nodes
+	availableNodes.Items = append(availableNodes.Items[:selectedNodeIndex], availableNodes.Items[selectedNodeIndex+1:]...)
```

Comment:
you can't delete the item from the slice and then try to return in the line below

Reply:
Is this a comment to the wrong comment("// Delete the selected node from the list of available nodes")?
Because the above codes just try to recreate a slice without the element of selectedNodeIndex.

Reply:
Hmm I'll try to explain how I understand Marc's comment with an example.

- Let's assume that the `availableNodes` contains only one node.
- selectedNodeIndex would be 0.
- after execution of 
```
availableNodes.Items = append(availableNodes.Items[:selectedNodeIndex], availableNodes.Items[selectedNodeIndex+1:]...)
```
availableNodes.Items would be empty
- `return &availableNodes.Items[selectedNodeIndex]` would cause a nil panic because it tries to reference a node that was removed from the slice

Reply:
@mshitrit @slintes Thank you for the explanation and correcting me.
I got the point  and fixed it with the following:
```
       // Randomly select a worker node
        selectedNodeIndex := r.Intn(len(availableNodes.Items))
+       selectedNode := availableNodes.Items[selectedNodeIndex]
        // Delete the selected node from the list of available nodes
        availableNodes.Items = append(availableNodes.Items[:selectedNodeIndex], availableNodes.Items[selectedNodeIndex+1:]...)
-       return &availableNodes.Items[selectedNodeIndex]
+       return &selectedNode
```